### ISSUE TITLE
Release a zio-http-test module with test helpers

### ISSUE THREAD

**Issue by loicdescotte** [dream11/zio-http#198]
It would be nice if the existing `zhttp.service.HttpRunnableSpec` (currenlty in zio-http test directory) was available for users, for example in a `zio-http-test` module.

So it would possible for users to write tests like : 

```scala
testM("test response") {
    val path = Root / "api" / "hello"
    for {
     a1 <- assertM(status(path))(equalTo(Status.OK))
     a2 <- assertM (request(path).map(r => responseAsString(r)))(equalTo("Hello world"))
    } yield a1 && a2
}
```

With `responseAsString` looking to something like : 

```scala
def responseAsString(reponse: UHttpResponse): String = {
    reponse.content match {
      case CompleteData(data) => new String(data.toArray)
      case StreamData(data) => data.run(Sink.foldLeft("")(???)) 
      case Empty => ""
    }
  }
```

**Comment by legopiraat** [comment]
Please! This would be amazing

**Comment by rleibman** [comment]
Take a look at https://github.com/dream11/zio-http/pull/308

**Comment by sviezypan** [comment]
hello @tusharmath Is this planned is some future release? 
Importing zhttp-test only brings [HttpWithTest](https://github.com/dream11/zio-http/blob/main/zio-http-test/src/main/scala/zhttp/test/test.scala) which doesn't do much.
I would love to use `HttpRunnableSpec` and others ....

**Comment by soujiro32167** [comment]
here is what I have so far:
```scala
trait ZioHttpTest {
 implicit class RichResponse[-R, +E](self: Response[R, E]) {
    def httpOrDie: UIO[HttpResponse[R, E]] = self match {
      case http: HttpResponse[R, E] => ZIO.succeed(http)
      case _                        => ZIO.die(new RuntimeException("only http requests are supported"))
    }
  }

  implicit class RichHttpResponse[-R, +E](self: HttpResponse[R, E]) {
    def bytes: ZIO[R, E, Chunk[Byte]] = self.content match {
      case HttpData.Empty              => ZIO.succeed(Chunk.empty)
      case HttpData.CompleteData(data) => ZIO.succeed(data)
      case HttpData.StreamData(data)   => data.runCollect
    }

    def string: ZIO[R, E, String] =
      bytes.map(c => new String(c.toArray))

    def fromJson[A: JsonDecoder]: ZIO[R, E, A] =
      string.flatMap(s => ZIO.fromEither(s.fromJson).orDieWith(new RuntimeException(_)))
  }

}
```

**Comment by soujiro32167** [comment]
Not sure how to compose errors in `fromJson` 🤔

**Comment by soujiro32167** [comment]
Ok, I made a mistake: I was using Tapir + zio-http - thats why I got an http app with `Response[R, Throwable]` as output and the need for this thing...

**Comment by amitksingh1490** [comment]
Released since 1.0.0.0-RC18
